### DISCUSSION TITLE
Followed "First Run Grbl Settings" - Still in Alarm state

### DISCUSSION THREAD

**Discussion by jerseyguy1996** [terjeio/grblHAL#227] [Q&A] [answered]
Trying every iteration I can think of here and I can't seem to get out of an Alarm state.  This is using the Teensy4.1 with nothing wired up to it yet.  I inverted control and limit signals.  I'm actually not sure why I needed to do that because I also don't have hard or soft limits enabled and DEFAULT_CHECK_LIMITS_AT_INIT is commented out.  Below is status and current config.  Any help is greatly appreciated!

`<Alarm|MPos:0.000,0.000,0.000|Bf:255,1023|FS:0,0|Pn:R|Ov:100,100,100>
ok`

`$$
$0=10.0
$1=25
$2=0
$3=0
$4=0
$5=7
$6=0
$10=511
$11=0.010
$12=0.002
$13=0
$14=14
$15=0
$16=0
$17=0
$18=0
$19=0
$20=0
$21=0
$22=0
$23=0
$24=25.000
$25=500.000
$26=250
$27=1.000
$28=0.100
$29=0.0
$30=100.
$31=0.
$32=1
$33=5000.000
$34=0.000
$35=0.000
$36=100.000
$37=0
$39=1
$40=0
$41=0
$42=2
$43=1
$44=4
$45=3
$46=0
$56=5.000
$57=100.000
$58=-5.000
$59=500.000
$60=0
$61=0
$62=0
$63=2
$64=0
$65=0
$100=157.635
$101=157.635
$102=250.000
$110=15000.000
$111=15000.000
$112=500.000
$120=15.000
$121=15.000
$122=10.000
$130=200.000
$131=200.000
$132=200.000
$341=0
$342=30.0
$343=25.0
$344=200.0`

**Comment by terjeio** [comment, marked as answer]
Try with $14=65 and $10=1535, the last one will add the alarm number to the status report.

**Reply by jerseyguy1996** [reply]
This was the result of both of those:
(Edit: Not sure why the carriage returns aren't working in the code block below)

`ok
$10=1535
ok
?
<Alarm:0|MPos:0.000,0.000,0.000|Bf:255,1023|FS:0,0|Pn:R>
ok
$14=65
ok
?
<Alarm:0|MPos:0.000,0.000,0.000|Bf:255,1023|FS:0,0|Pn:HS>
ok
$14=78
ok
?
<Alarm:0|MPos:0.000,0.000,0.000|Bf:255,1023|FS:0,0|Pn:R>
ok`

**Comment by phil-barrett** [comment]
Did you try $14=78 as suggested in the Wiki entry? Also, are you using ioSender?  Very helpful in showing what the inputs are doing.

This probably won't give you much additional insight but for others that want to use a bare Teensy [this blog may be helpful](https://www.grbl.org/single-post/naked-teensy-party).

**Reply by jerseyguy1996** [reply]
Yes I tried that (see my reply to terjeio above).  I'm using Lightburn but it's just in the console window.

**Reply by terjeio** [reply]
What is the output of `$I`?
Are you compiling yourself or do you use precompiled firmware?

**Reply by jerseyguy1996** [reply]
I'm compiling myself.  This is the result of $I:

`ok
$I
[VER:1.1f(IMXRT1062).20201103:]
[OPT:VNMSL,1023,1024,3,0]
Target buffer size found
[NEWOPT:*FLASH,TC]
[DRIVER VERSION:201014]
[DRIVER OPTIONS:USB.2]
ok`

**Comment by phil-barrett** [comment]
Hmmm, just tried lightburn with a naked teensy.  Set $14=78 outside of lightburn as it does not recognize the enhanced settings of grblHAL. [edit: I was wrong, lightburn **does** allow $14=78 in the Console, not sure why I got an error earlier] Here is a screen shot of it running a simple job:
![lightburn naked teensy](https://user-images.githubusercontent.com/16570958/106390792-05038a00-639f-11eb-8fcd-84006f928b01.png)

**Reply by jerseyguy1996** [reply]
Oh that's interesting.....$14=78 followed by $$ and it looks like $14 is set to 14.  I tried that using the Arduino IDE Serial Monitor and using the Lightburn console window.  What gives?

**Reply by phil-barrett** [reply]
There appears to be some oddness going on. I did get LB to accept $14=78 and it took.  Edited my post above.  Anyway, are you up and running? (well as up and running as you can be on a naked teensy)

**Reply by terjeio** [reply]
Can you try with the [test branch](https://github.com/terjeio/grblHAL/tree/test)?

> Oh that's interesting.....$14=78 followed by $$ and it looks like $14 is set to 14.

In the latest version to be uploaded it will even be worse as inversion and disable pullup masks for input pins not available will be discarded. This to make driver code simpler.

**Reply by jerseyguy1996** [reply]
Sadly no....still getting the same issue.  I set some of the defaults in my config.h.  I'm going to roll everything back to the original settings and see if I can get that working first.  Thanks for being so quick to reply to all of this!  I'll be back once I back out all of my config changes.

**Reply by jerseyguy1996** [reply]
> Can you try with the [test branch](https://github.com/terjeio/grblHAL/tree/test)?
> 
> > Oh that's interesting.....$14=78 followed by $$ and it looks like $14 is set to 14.
> 
> In the latest version to be uploaded it will even be worse as inversion and disable pullup masks for input pins not available will be discarded. This to make driver code simpler.

Yep I'll give that a try

**Reply by phil-barrett** [reply]
Also, in my_machine.h, make sure you have 
`#define BOARD_T41U5XBB
`
(at or near line 27).  Not strictly necessary but will make sure we are on the same page.

**Reply by terjeio** [reply]
> Also, in my_machine.h, make sure you have
#define BOARD_T41U5XBB 

Yes I saw that, it could be an issue with the generic map. I has the halt input routed to Reset, not EStop.

**Reply by jerseyguy1996** [reply]
@terjeio @phil-barrett Okay........started fresh using the test branch...didn't change a single thing except set $14=78 (which finally took) and after a power cycle it is working:-)  I'll start enabling things one at a time and see if anything breaks it.  I'll let you know how it goes.  Thanks both of you so much for your assistance and for creating and maintaining such an amazing piece of software!  Do you guys have a foundation or paypal that I can kick some bucks your way to do my part in helping to support your work?

**Reply by jerseyguy1996** [reply]
Hmmmm......first change was trying to enable laser mode.  Used $32=1 but it doesn't seem to want to stick.  $$ results in $32=0.  Tried changing it in Lightburn and in the arduino ide serial monitor.  Could this be something that was done to affect the test branch?

**Reply by phil-barrett** [reply]
Glad you are moving forward now.  

Nothing formalized. There is no donation mechanism, so far... I sell a T4.1 breakout board and use the profits to make new designs, pay for grbl.org, buy components to test against and so on.

**Reply by phil-barrett** [reply]
We crossed in the mail.  Hmmm, I am able to send $32=1 in the Console and it seems to stick ($$ shows it set to 0).  I'm using the main branch. I can switch $32 back and forth.  Do you have eeprom enabled?

**Reply by terjeio** [reply]
> Hmmmm......first change was trying to enable laser mode. Used $32=1 but it doesn't seem to want to stick. $$ results in $32=0. Tried changing it in Lightburn and in the arduino ide serial monitor. Could this be something that was done to affect the test branch?

Oops, I have forgotten to update for that - it is a regression from the large settings susbsystem overhaul I have been working on. I'll commit a fix soon.

**Reply by jerseyguy1996** [reply]
> We crossed in the mail. Hmmm, I am able to send $32=1 in the Console and it seems to stick ($$ shows it set to 0). I'm using the main branch. I can switch $32 back and forth. Do you have eeprom enabled?

Shouldn't $$ respond with $32=1 after setting it?  Since it's just a bare board right now, I don't have any external eeprom attached.  I was under the impression that the Teensy4.1 reserved a portion of flash to store settings that are expected to persist.  Looks like the EEPROM_ENABLE in my_machine.h was for external eeprom.

**Reply by jerseyguy1996** [reply]
oops @terjeio looks like we crossed our posts.  I'll switch back to the master branch

**Reply by phil-barrett** [reply]
eeprom should not be enabled for a naked teensy.  not sure if that is the regression terje was referring to.

**Reply by jerseyguy1996** [reply]
Okay back on the Master branch and we are working again!  Thanks again for all of your help!  I'll keep enabling things and let you know how it goes:-)

**Reply by terjeio** [reply]
> eeprom should not be enabled for a naked teensy. not sure if that is the regression terje was referring to.

Nope - the regression was bad code on my part.

Test branch has now been updated with a fix.

>  Do you guys have a foundation or paypal that I can kick some bucks your way to do my part in helping to support your work?

I do not have any set up yet - don't know if the local tax man will take an interest in that. What would please me if users could donate time to help out with testing the test brach - to be renamed to "Edge" soon.

**Comment by Shiraj123** [comment]
follow me I'll follow back

**Comment by jerseyguy1996** [comment]
@terjeio @phil-barrett just to close the loop on this, I wired the Teensy up to the K40 today and it is working like a charm :-).  Ran some large raster images through it that used to choke up the Atmega328 running classic GRBL and even at 250mm/s it wasn't slowing down at all in the complex parts.  I am very happy and thank you for putting in so much time and effort into creating such an outstanding piece of work!  I'm having one spurious issue that keeps popping up but I'll start a new discussion as it doesn't relate to this issue.
